lib/capitano.js

```diff
@@ -0,0 +1,188 @@
+        return {name: app.app_name, value: app}
+      })
+    })
+  })
+}
+
+function selectApplicationCommit (resin, form, appId) {
+  return resin.models.build.getAllByApplication(appId)
+  .then(function (builds) {
+    if (builds.length === 0) {
+      throw new Error('This application has no builds')
+    }
+    builds = builds.filter(function (build) {
+      return build.status === 'success'
+    }).sort(function (a, b) {
+      return -cmp(a.push_timestamp, b.push_timestamp)
+    })
+    const choices = builds.map(function (build) {
+      return {
+        name: build.push_timestamp + ' - ' + build.commit_hash,
+        value: build.commit_hash
+      }
+    })
+    choices.splice(0, 0, {'name': LATEST, 'value': LATEST})
+    return form.ask({
+      message: 'Select a build',
+      type: 'list',
+      default: LATEST,
+      choices: choices
+    })
+  })
+}
+
+function disableAutomaticUpdates (form, resin, application, commit) {
+  if ((commit === LATEST) || !application.should_track_latest_release) {
+    return Promise.resolve()
+  }
+  const message = (
+    'This application is set to automatically update all devices to the ' +
+    'latest available version.\n' +
+    'You need to disable this if you want to preload a specific build. ' +
+    'Do you want to disable automatic updates for this application?'
+  )
+  return form.ask({
+    message: message,
+    type: 'list',
+    default: true,
+    choices: [
+      {name: 'Yes', value: true},
+      {name: 'No', value: false}
+    ]
+  })
+  .then(function (update) {
+    if (!update) {
+      return
+    }
+    return resin.pine.patch({
+      resource: 'application',
+      id: application.id,
+      body: {
+        should_track_latest_release: false
+      }
+    })
+  })
+}
+
+module.exports = {
+  signature: 'preload <image>',
+  description: '(beta) preload an app on a disk image',
+  help: `Warning: 'resin preload' requires Docker to be correctly installed in
+your shell environment. For more information (including Windows support)
+please check the README here: https://github.com/resin-io/resin-cli .
+
+Use this command to preload an application to a local disk image.
+
+Examples:
+  $ resin preload resin.img --app 12345 --commit e1f2592fc6ee949e68756d4f4a48e49bff8d72a0 --splash-image some-image.png
+  $ resin preload resin.img`,
+  permission: 'user',
+  primary: true,
+  options: [
+    {
+      signature: 'app',
+      parameter: 'appId',
+      description: 'id of the application to preload',
+      alias: 'a'
+    },
+    {
+      signature: 'commit',
+      parameter: 'hash',
+      description: 'a specific application commit to preload (ignored if no appId is given)',
+      alias: 'c'
+    },
+    {
+      signature: 'splash-image',
+      parameter: 'splashImage.png',
+      description: 'path to a png image to replace the splash screen',
+      alias: 's'
+    },
+    {
+      signature: 'dont-detect-flasher-type-images',
```

Comment:
This option feels a bit verbose, I think at the very least we should add an alias for it,